rust/src/store/best_ledger_store_impl.rs

```diff
@@ -144,7 +152,15 @@ impl BestLedgerStore for IndexerStore {
                     diff.map(|d| {
                         (
                             d.account_diffs.into_iter().flatten().collect(),
-                            d.new_pk_balances.into_keys().collect(),
+                            d.new_pk_balances
+                                .into_iter()
+                                .flat_map(|(pk, tokens)| {
+                                    tokens
+                                        .into_keys()
+                                        .map(|token| (pk.to_owned(), token))
+                                        .collect::<HashSet<_>>()
+                                })
+                                .collect(),
```

Comment:
nit: This looks like a dupe of lines 134-142. Perhaps it could be combined into some sort of function.